satellite/satellitedb/overlaycache.go

```diff
@@ -1338,3 +1338,130 @@ func populateUpdateFields(dbNode *dbx.Node, updateReq *overlay.UpdateRequest) db
+			Type:     pb.NodeType_STORAGE,
+		}
+		_, err = cache.UpdateNodeInfo(ctx, node.NodeID, &pbInfo)
+		if err != nil {
+			return Error.Wrap(err)
+		}
+	case *pq.Driver:
+		// v is a single feedback value that allows us to update both alpha and beta
+		var v float64 = -1
+		if node.IsUp {
+			v = 1
+		}
+		uptimeReputationAlpha := config.UptimeReputationLambda*config.UptimeReputationAlpha0 + config.UptimeReputationWeight*(1+v)/2
+		uptimeReputationBeta := config.UptimeReputationLambda*config.UptimeReputationBeta0 + config.UptimeReputationWeight*(1-v)/2
+		semVer, err := version.NewSemVer(node.Version.GetVersion())
+		if err != nil {
+			return Error.New("unable to convert version to semVer")
+		}
+		start := time.Now()
+		query := `
+			INSERT INTO nodes
+			(
+				id, address, last_net, protocol, type,
+				email, wallet, free_bandwidth, free_disk,
+				uptime_success_count, total_uptime_count, 
+				last_contact_success,
+				last_contact_failure,
+				audit_reputation_alpha, audit_reputation_beta, uptime_reputation_alpha, uptime_reputation_beta,
+				major, minor, patch, hash, timestamp, release
+			)
+			VALUES (
+				$1, $2, $3, $4, $5,
+				$6, $7, $8, $9,
+				$10::bool::int, 1,
+				CASE WHEN $10 IS TRUE THEN current_timestamp
+					ELSE '0001-01-01 00:00:00+00'
+				END,
+				CASE WHEN $10 IS FALSE THEN current_timestamp
+					ELSE '0001-01-01 00:00:00+00'
+				END,
+				$11, $12, $13, $14,
+				$18, $19, $20, $21, $22, $23
+			)
+			ON CONFLICT (id)
+			DO UPDATE
+			SET
+				address=$2,
+				last_net=$3,
+				protocol=$4,
+				email=$6,
+				wallet=$7,
+				free_bandwidth=$8,
+				free_disk=$9,
+				total_uptime_count=nodes.total_uptime_count+1,
+				uptime_reputation_alpha=$16::numeric*nodes.uptime_reputation_alpha + $17::numeric*$10::bool::int,
+				uptime_reputation_beta=$16::numeric*nodes.uptime_reputation_beta + $17::numeric*(NOT $10)::bool::int,
+				uptime_success_count = nodes.uptime_success_count + $10::bool::int,
+				last_contact_success = CASE WHEN $10 IS TRUE
+					THEN current_timestamp
+					ELSE nodes.last_contact_success
+				END,
+				last_contact_failure = CASE WHEN $10 IS FALSE
+					THEN current_timestamp
+					ELSE nodes.last_contact_failure
+				END,
+				-- this disqualified case statement resolves to: 
+				-- when (new.uptime_reputation_alpha /(new.uptime_reputation_alpha + new.uptime_reputation_beta)) <= config.UptimeReputationDQ
+				disqualified = CASE WHEN (($16::numeric*nodes.uptime_reputation_alpha + $17::numeric*$10::bool::int) / (($16::numeric*nodes.uptime_reputation_alpha + $17::numeric*$10::bool::int) + ($16::numeric*nodes.uptime_reputation_beta + $17::numeric*(NOT $10)::bool::int))) <= $15 AND nodes.disqualified IS NULL
+					THEN current_timestamp
+					ELSE nodes.disqualified
+				END;
+			`
+		_, err = cache.db.ExecContext(ctx, query,
+			// args $1 - $5
+			node.NodeID.Bytes(), node.Address.GetAddress(), node.LastIP, node.Address.GetTransport(), int(pb.NodeType_STORAGE),
```

Comment:
I believe node.Address.GetTransport() will be nil. The contact endpoint is not setting this field before passing NodeCheckInInfo to this method

Reply:
oh yeah i forgot about that. i wonder if i should change it back to send the transport. Or if i should remove the transport here in the query. thoughts?

Reply:
I'm not sure why the transport is necessary, but I think passing it along is the safer bet

Reply:
Oh wait. That info isn't being sent by the node. But if it's a gRPC endpoint that is calling the overlay method, the transport must be gRPC right? I would add a Transport field to NodeCheckInInfo and set it to `pb.NodeTransport_TCP_TLS_GRPC` when passing it from the endpoint to the overlay

Reply:
Hmmm, then again we're basically hard coding this value everywhere. Hard code it, or pass it, up to you

Reply:
hmm. hard one. maybe i will hard code for now for simplicity.